### PR TITLE
CI: Add shell script linting

### PR DESCRIPTION
**Pull request by mikkopiu**
#### Summary
- CI: add shell script linting with shellcheck
   - Lint all shell scripts with the excellent shellcheck tool to avoid the most common mistakes (variable expansion, incorrect whitespace, unused variables etc)
   - Fix minor issues in existing scripts found by the new script
   - Ran in parallel to multiple slower jobs -> no increase in total workflow time
- Improve AWS metadata error logging in all image entrypoints
   - Don't suppress error messages from `curl` when fetching host IP from AWS metadata during image startups, while still falling back to a default value
   - Also ignore SC2155: we already handle the error case with the fallback value "UNAVAILABLE" and output error messages from `curl` to stderr (without crashing as that's not desirable here)
- Fix minor existing issues with all the shell scripts

### CODE REVIEW

File: service/entrypoint.sh [hunk 1]
```diff
@@ -7,7 +7,8 @@
 set -euo pipefail
 
 # For logback tagging (allow for local fallback)
-export HOST_IP=$(wget -qO- http://169.254.169.254/latest/meta-data/local-ipv4 || printf 'UNAVAILABLE')
+HOST_IP=$(curl --fail -s http://169.254.169.254/latest/meta-data/local-ipv4 || printf 'UNAVAILABLE')
+export HOST_IP
```
Comment (ilja-git) [inline service/entrypoint.sh:11]: Is separate export required as we use "||"? We might want to use -sSf also o get the error message to stderr for debugging purposes.

Comment (ilja-git) [reply service/entrypoint.sh:11]: If the separate export is because linting then ignore me.

**Review by ilja-git — APPROVED** [review]

File: service/entrypoint.sh (hunk 1 above)
Comment (mikkopiu) [reply service/entrypoint.sh:11]: It was indeed because of the linter but you make a good point, I'll fix these.

**Review by ilja-git — APPROVED** [review]
LGTM

**Pull request merged**